pkg/odo/cli/component/describe.go

```diff
@@ -35,15 +41,55 @@ var describeCmd = &cobra.Command{
 		componentDesc, err := component.GetComponentDesc(client, componentName, applicationName)
 		odoutil.LogErrorAndExit(err, "")
 
-		odoutil.PrintComponentInfo(componentName, componentDesc)
+		if outputFlag == "json" {
+			componentDef := getMachineReadableFormat(componentDesc)
+			out, err := json.Marshal(componentDef)
+			odoutil.LogErrorAndExit(err, "")
+			fmt.Println(string(out))
+		} else {
+
+			odoutil.PrintComponentInfo(componentName, componentDesc)
+		}
 	},
 }
 
+func getMachineReadableFormat(componentDesc component.Description) component.Component {
+	var urls []string
+	for _, url := range componentDesc.URLs {
+		urls = append(urls, url.Name)
+	}
+
+	var storage []string
+	for _, store := range componentDesc.Storage {
+		storage = append(storage, store.Name)
+	}
+
+	componentDef := component.Component{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Component",
+			APIVersion: "odo.openshift.io/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: componentDesc.ComponentName,
+		},
+		Spec: component.ComponentSpec{
+			Type:    componentDesc.ComponentImageType,
+			Source:  componentDesc.Path,
+			URL:     urls,
+			Storage: storage,
+			Env:     componentDesc.Env,
```

Comment:
what is this env supposed to be?
I am sure I have some env vars injected into my component container but I see the below:
```
[anmolbabu@localhost odo-samples]$ odo describe -o json
{"kind":"Component","apiVersion":"odo.openshift.io/v1beta1","metadata":{"name":"nodejs-ex-nodejs-beov","creationTimestamp":null},"spec":{"type":"nodejs","source":"file:///home/anmolbabu/git-srcs/src/github.com/redhat-developer/odo-samples/source/nodejs-ex","url":["nodejs-ex-nodejs-beov-8080"],"Env":null}}
```

Reply:
It seems even the `odo describe` doesn't provide any env vars

Reply:
when you do, 

```
odo create nodejs --env <a=b>
```

so values provided by users as env

Reply:
@anmolbabu Do you think that internal env variables that are automatically injected should be here?